frontend/src/metabase/components/DateAllOptions/DateAllOptions.unit.spec.tsx

```diff
@@ -0,0 +1,159 @@
+import userEvent from "@testing-library/user-event";
+import { useState } from "react";
+import { renderWithProviders, screen } from "__support__/ui";
+import type { DateAllOptionsProps } from "metabase/components/DateAllOptions/DateAllOptions";
+import { DateAllOptions } from "metabase/components/DateAllOptions/DateAllOptions";
+import type { DateShortcutOptions } from "metabase/query_builder/components/filters/pickers/DatePicker/DatePickerShortcutOptions";
+import { DATE_SHORTCUT_OPTIONS } from "metabase/query_builder/components/filters/pickers/DatePicker/DatePickerShortcutOptions";
+
+const TEST_DAY_MONTH_OPTIONS: Record<string, string> = {
+  Today: "thisday",
+  Yesterday: "past1days",
+  "Last Week": "past1weeks",
+  "Last 7 Days": "past7days",
+  "Last 30 Days": "past30days",
+  "Last Month": "past1months",
+  "Last 3 Months": "past3months",
+  "Last 12 Months": "past12months",
+};
+
+const TEST_MISC_OPTIONS = ["Specific dates…", "Relative dates…", "Exclude…"];
+
+const TEST_DAY_MONTH_SHORTCUTS = Object.keys(TEST_DAY_MONTH_OPTIONS);
+
+const filterShortcut = (filterValue: string): DateShortcutOptions => {
+  return {
+    DAY_OPTIONS: DATE_SHORTCUT_OPTIONS.DAY_OPTIONS.filter(
+      ({ displayName }) => displayName !== filterValue,
+    ),
+    MONTH_OPTIONS: DATE_SHORTCUT_OPTIONS.MONTH_OPTIONS.filter(
+      ({ displayName }) => displayName !== filterValue,
+    ),
+    MISC_OPTIONS: DATE_SHORTCUT_OPTIONS.MISC_OPTIONS.filter(
+      ({ displayName }) => displayName !== filterValue,
+    ),
+  };
+};
+
+const TestDateAllOptions = ({
+  dateShortcutOptions,
+  onClose,
+  setValue,
+}: DateAllOptionsProps) => {
+  const [testValue, setTestValue] = useState<string | null>(null);
+
+  const onSetValue = (val: string | null) => {
+    setTestValue(val);
+    setValue(val);
+  };
+
+  return (
+    <DateAllOptions
+      value={testValue ?? undefined}
+      setValue={onSetValue}
+      onClose={onClose}
+      disableOperatorSelection={false}
+      dateShortcutOptions={dateShortcutOptions}
+    />
+  );
+};
+
+const setup = ({
+  value = undefined,
+  dateShortcutOptions = undefined,
+}: {
+  value?: string;
+  dateShortcutOptions?: DateShortcutOptions;
+} = {}) => {
+  const setValueMock = jest.fn();
+  const onCloseMock = jest.fn();
+
+  renderWithProviders(
+    <TestDateAllOptions
+      setValue={setValueMock}
+      onClose={onCloseMock}
+      value={value}
+      dateShortcutOptions={dateShortcutOptions}
+    />,
+  );
+
+  return {
+    setValueMock,
+    onCloseMock,
+  };
+};
+
+describe("DateAllOptions", () => {
```

Comment:
`DatePicker` has the tests for the application and navigation within the `DatePicker` component, so we're just testing the exclusion of shortcuts, and that input and output from the `DateAllOptions` widgets works as expected.